File: RStudio/src/select-origin.py
# Filters the dataset to select an origin airport
#
# To run as a renku workflow step:
# renku run python src/select-origin.py --name filter --input data/2019-01_us_fli_1.0/2019-01-flights.csv.zip

import pandas as pd
import sys

origin = "BOS"

df = pd.read_csv("data/2019-01_us_fli_1.0/2019-01-flights.csv.zip")
df.head()

is_origin =  df['ORIGIN'] == origin

df_origin = df[is_origin]

df_origin.head()

df_origin.to_csv('output/filtered.csv')

File: RStudio/src/select-origin.R
# Filters the dataset to select an origin airport
#
# To run as a renku workflow step:
# $ renku run Rscript src/select-origin.R --input data/2019-01_us_fli_1.0/2019-01-flights.csv.zip

library(readr)
library(dplyr)

origin <- "BOS"

df <- read_csv(unzip("data/2019-01_us_fli_1.0/2019-01-flights.csv.zip", exdir="/tmp"),
               col_types = cols(.default=col_guess(), CRS_DEP_TIME="I", DEP_TIME="I"))

df_origin <- filter(df, ORIGIN == origin)

write_csv(df_origin, "output/filtered.csv")
